### PR TITLE
Sort the global list of dependencies

### PR DESCRIPTION
**Pull request by elharo**
@garrettjonesgoogle @suztomo Some of the things we pull in are quite surprising. E.g Avalon?!

We may want to add some exploratory tools to this so we can easily dig in and find out where detritus like this is coming from.

### CODE REVIEW

File: dashboard/src/test/java/com/google/cloud/tools/opensource/dashboard/DashboardTest.java [hunk 1]
```diff
@@ -121,10 +124,22 @@ public void testDashboard()
...
       for (int i = 0; i < li.size(); i++) {
         String coordinates = li.get(i).getValue();
         // fails if these are not valid Maven coordinates
         new DefaultArtifact(coordinates);
+        coordinateList.add(coordinates);
+      }
+      
+      ArrayList<String> sorted = new ArrayList<>(coordinateList);
+      Comparator<String> comparator = new SortWithoutVersion();
+      Collections.sort(sorted, comparator );
```
Comment (suztomo) [inline dashboard/src/test/java/com/google/cloud/tools/opensource/dashboard/DashboardTest.java]: Unnecessary space after `comparator`

File: dashboard/src/test/java/com/google/cloud/tools/opensource/dashboard/DashboardTest.java [hunk 2]
```diff
@@ -121,10 +124,22 @@ public void testDashboard()
...
+      
+      for (int i = 0; i < sorted.size(); i++) {
+        Assert.assertEquals("Coordinates are not sorted: ", sorted.get(i),
+            coordinateList.get(i));
```
Comment (garrettjonesgoogle) [inline dashboard/src/test/java/com/google/cloud/tools/opensource/dashboard/DashboardTest.java]: It seems like we could have much more concise assertions by using the `Truth` library

File: dashboard/src/test/java/com/google/cloud/tools/opensource/dashboard/DashboardTest.java (hunk 1 above)
Comment (elharo) [reply dashboard/src/test/java/com/google/cloud/tools/opensource/dashboard/DashboardTest.java]: fixed

File: dashboard/src/test/java/com/google/cloud/tools/opensource/dashboard/DashboardTest.java (hunk 2 above)
Comment (elharo) [reply dashboard/src/test/java/com/google/cloud/tools/opensource/dashboard/DashboardTest.java]: Specific suggestions? I looked there and didn't find anything on point. I did have to replace a general equals check on the list with an iterated check on each element to find the exact elements that were sorted out of order when my test failed.

Comment (garrettjonesgoogle) [reply dashboard/src/test/java/com/google/cloud/tools/opensource/dashboard/DashboardTest.java]: I think you have to combine a couple things, like `containsExactly` and `inOrder`. I'm not sure what the error messaging looks like when it fails.

Consider this optional if you want to look into it.

Comment (suztomo) [reply dashboard/src/test/java/com/google/cloud/tools/opensource/dashboard/DashboardTest.java]: `Truth.assertThat(coordinateList).containsExactlyElementsIn(sorted).inOrder();`?
http://google.github.io/truth/api/latest/com/google/common/truth/Ordered.html

**Pull request merged**